Makefile

```diff
@@ -83,18 +83,36 @@ help:/
           } \
         }' $(MAKEFILE_LIST)
 
+GOFORMAT_FILES := $(shell find  . -name '*.go' \
+	! -wholename './bindata_assetfs.go' \
+	! -wholename './migration/sqlbindata_test.go' \
+	! -wholename './migration/sqlbindata.go' \
+	! -path './vendor/*' \
+	! -path './app/*' \
+	! -path './client/*' \
+	! -path './tool/cli/*' )
+
 .PHONY: check-go-format
 ## Exists with an error if there are files whose formatting differs from gofmt's
 check-go-format: prebuild-check
-	@gofmt -s -l ${SOURCES} 2>&1 \
+	@gofmt -s -l ${GOFORMAT_FILES} 2>&1 \
 		| tee /tmp/gofmt-errors \
 		| read \
 	&& echo "ERROR: These files differ from gofmt's style (run 'make format-go-code' to fix this):" \
 	&& cat /tmp/gofmt-errors \
 	&& exit 1 \
 	|| true
 
+.PHONY: install-git-hooks
+## Creates useful git hooks (e.g. pre-push go-format-code check)
+install-git-hooks:
+	ln -sfv ../../.pre-push .git/hooks/pre-push
 
+CLEAN_TARGETS += clean-git-hooks
```

Comment:
Not sure if this makes sense as part of the normal CLEAN_TARGETS

As it will kill any other pre-push hooks that are installed regardless of it being this hook or not.